### PR TITLE
Resolve #1085: Free __DEV__ variable provided by webpack should be used across RSK i…

### LINKED ISSUE #1085: Free __DEV__ variable provided by webpack should be used instead of NODE_ENV checks
To my knowledge, `__DEV__` is a facebook code idiom that showed up relatively recently in the main React repo. Can see it all over the place [here](https://github.com/facebook/react/search?utf8=%E2%9C%93&q=__DEV__) (may need to login to see github search results). Their build process has a step that replaces `__DEV__` with `process.env.NODE_ENV !== 'production'`. 

RSK currently has something similar, `__DEV__` is made available via webpack, we just have not been using it. This is likely due to a missing exception in the `.eslintrc` allowing a `no-underscore-dangle` exception for `__DEV__` (and lack awareness of availability too!).

Can see where `__DEV__` is made available in the current [webpack.config.js](https://github.com/kriasoft/react-starter-kit/blob/master/tools/webpack.config.js#L239) here (same for client/server):
```
// tools/webpack.config.js
... 
   // Define free variables
    // https://webpack.github.io/docs/list-of-plugins.html#defineplugin
    new webpack.DefinePlugin({
      'process.env.NODE_ENV': isDebug ? '"development"' : '"production"',
      'process.env.BROWSER': true,
      __DEV__: isDebug,
    }),
...
```

Pull request would replace both process.env.NODE_ENV !== 'production' statements with `__DEV__` and add a `no-comma-dangle` exception to the `.eslintrc`. `isDebug` in `webpack.config.js` should also be converted into `__DEV__` as there is no reason to violate convention.

++Readability

(p.s. Sorry for the pull request spam.. `eslint --fix` which runs precommit alters code WHILE committing. So `__DEV__ = true` if consumed like `{option1: 'foo', __DEV__ : __DEV__}` is transformed and committed as `{option1: 'foo', __DEV__}` which is less readable and certainly not a norm.. have never seen this pattern. Honestly did not even know `var x = true; var y = {x};` was valid JS. Completely unexpected behaviour. Ended up disabling es-lint for those lines)

### PR DESCRIPTION
**Pull request by buildbreakdo**
Resolve #1085 
To my knowledge, `__DEV__` is a facebook code idiom that showed up relatively recently in the main React repo. Can see it all over the place [here](https://github.com/facebook/react/search?utf8=%E2%9C%93&q=__DEV__) (may need to login to see github search results). Their build process has a step that replaces `__DEV__` with `process.env.NODE_ENV !== 'production'`. 

RSK currently has something similar, `__DEV__` is made available via webpack, we just have not been using it. This is likely due to a missing exception in the `.eslintrc` allowing a `no-underscore-dangle` exception for `__DEV__` (and lack awareness of availability too!).

Can see where `__DEV__` is made available in the current [webpack.config.js](https://github.com/kriasoft/react-starter-kit/blob/master/tools/webpack.config.js#L239) here (same for client/server):
```
// tools/webpack.config.js
... 
   // Define free variables
    // https://webpack.github.io/docs/list-of-plugins.html#defineplugin
    new webpack.DefinePlugin({
      'process.env.NODE_ENV': isDebug ? '"development"' : '"production"',
      'process.env.BROWSER': true,
      __DEV__: isDebug,
    }),
...
```

Pull request would replace both process.env.NODE_ENV !== 'production' statements with `__DEV__` and add a `no-comma-dangle` exception to the `.eslintrc`. `isDebug` in `webpack.config.js` should also be converted into `__DEV__` as there is no reason to violate convention.

++Readability

(p.s. Sorry for the pull request spam.. `eslint --fix` which runs precommit alters code WHILE committing. So `__DEV__ = true` if consumed like `{option1: 'foo', __DEV__ : __DEV__}` is transformed and committed as `{option1: 'foo', __DEV__}` which is less readable and certainly not a norm.. have never seen this pattern. Honestly did not even know `var x = true; var y = {x};` was valid JS. Completely unexpected behaviour. Ended up disabling es-lint for those lines)

### CODE REVIEW

File: tools/webpack.config.js [hunk 1]
```diff
@@ -234,9 +234,9 @@ const clientConfig = extend(true, {}, config, {
     // Define free variables
     // https://webpack.github.io/docs/list-of-plugins.html#defineplugin
     new webpack.DefinePlugin({
-      'process.env.NODE_ENV': isDebug ? '"development"' : '"production"',
+      'process.env.NODE_ENV': __DEV__ ? '"development"' : '"production"',
       'process.env.BROWSER': true,
-      __DEV__: isDebug,
+      __DEV__: __DEV__, // eslint-disable-line
```
Comment (langpavel) [inline tools/webpack.config.js]:
```diff
-      __DEV__: __DEV__, // eslint-disable-line
+      __DEV__,
```

**Comment by buildbreakdo** [comment]
@langpavel Thanks for the review, merged with latest webpack v2.2.1 changes and removed the `eslint-disable-line`. Cheers

**Comment by buildbreakdo** [comment]
@langpavel Oops -- forgot to remove the same line from the server config. Done now 👍

**Comment by langpavel** [comment]
Rebased in #1114.

**Pull request closed without merging**